### PR TITLE
Inheritance mode support for pull command

### PR DESCRIPTION
**Pull request by dragosv**
This is needed so that only translated strings are brought back, otherwise, the integration between Mojito and serge.io (Open-Source Solution for Continuous Localization by Evernote: https://serge.io) will not work properly.

See the plugin at  https://github.com/dragosv/serge/tree/mojito, will need to add this extra parameter once the pull request is accepted. The cli adds --inheritance-mode as a parameter for the pull command and the rest api will default to USE_PARENT if none specified. Also, the serge.io plugin should use REMOVE_UNTRANSLATED.

The plugin will be pushed into its own repository that I will support, as serge.io will not support any other translation services outside Pootle.

### CODE REVIEW

File: cli/src/main/java/com/box/l10n/mojito/cli/command/PullCommand.java [hunk 1]
```diff
@@ -62,6 +60,11 @@
     @Parameter(names = {Param.SOURCE_REGEX_LONG, Param.SOURCE_REGEX_SHORT}, arity = 1, required = false, description = Param.SOURCE_REGEX_DESCRIPTION)
     String sourcePathFilterRegex;
 
+    @Parameter(names = {"--inheritance-mode"}, required = false, description = "How to handle missing translations "
+            + "the parent (USE_PARENT for using parent). ",
+            converter = LocalizedAssetBodyInheritanceMode.class)
+    LocalizedAssetBody.InheritanceMode inheritanceMode = null;
```
Comment (aurambaj) [inline cli/src/main/java/com/box/l10n/mojito/cli/command/PullCommand.java]: Initialize with default value: USE_PARENT. That way the documentation shows the default value

File: cli/src/main/java/com/box/l10n/mojito/cli/command/PullCommand.java [hunk 2]
```diff
@@ -62,6 +60,11 @@
     @Parameter(names = {Param.SOURCE_REGEX_LONG, Param.SOURCE_REGEX_SHORT}, arity = 1, required = false, description = Param.SOURCE_REGEX_DESCRIPTION)
     String sourcePathFilterRegex;
 
+    @Parameter(names = {"--inheritance-mode"}, required = false, description = "How to handle missing translations "
```
Comment (aurambaj) [inline cli/src/main/java/com/box/l10n/mojito/cli/command/PullCommand.java]: description -> `Inheritance Mode. Used when there is no translations in the target locale for a text unit. (USE_PARENT to fallback to parent locale translation, REMOVE_TRANSLATED to remove the text unit from the file)`

File: cli/src/main/java/com/box/l10n/mojito/cli/command/LocalizedAssetBodyInheritanceMode.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    @Override
+    public LocalizedAssetBody.InheritanceMode convert(String string) {
+
+        LocalizedAssetBody.InheritanceMode statusEqualTarget = null;
+
+        if (string != null) {
+            try {
+                statusEqualTarget = LocalizedAssetBody.InheritanceMode.valueOf(string.toUpperCase());
+            } catch (IllegalArgumentException iae) {
+                throw new ParameterException("Invalid \"inheritance type\" processing type [" + string + "]");
```
Comment (aurambaj) [inline cli/src/main/java/com/box/l10n/mojito/cli/command/LocalizedAssetBodyInheritanceMode.java]: `inheritance type` -> `inheritance mode`

File: webapp/src/test/java/com/box/l10n/mojito/service/tm/TMServiceTest.java [hunk 4]
```diff
@@ -1346,7 +1347,7 @@ public void testLocalizeXtb() throws Exception {
             logger.debug("source=[{}]", textUnitDTO.getSource());
         }
 
-        String localizedAsset = tmService.generateLocalized(asset, assetContent, repoLocale, "en-GB", null);
+        String localizedAsset = tmService.generateLocalized(asset, assetContent, repoLocale, "en-GB", null, InheritanceMode.USE_PARENT);
         logger.debug("localized=\n{}", localizedAsset);
         assertEquals(expectedContent, localizedAsset);
     }
```
[the remark this answers is not in the document]
Comment (dragosv) [reply webapp/src/test/java/com/box/l10n/mojito/service/tm/TMServiceTest.java]: Added test for a few formats. Works except for the xtb one as it fails to output translationbundle (ignored that test for now)

**Comment by aurambaj** [comment]
Thanks for all the tests! Let's keep `Xtb` the way you did with `@Ignore` shouldn't be a big deal.

If you can just fix the import. I'd rather stick the standard order and no change is needed in that case (sorry for being picky) and I'll merge!

**Comment by aurambaj** [comment]
@dragosv Thanks!!

**Comment by jeeyi** [comment]
@dragosv Awesome!! 👍 💯 🏆

**Pull request merged**